users/validators.py

```diff
@@ -23,3 +23,21 @@ def __call__(self, users_birth_date: Optional[datetime.date]) -> Optional[dateti
             if not self.MINIMAL_ACCETABLE_AGLE <= age <= self.MAXIMAL_ACCEPTABLE_AGE:
                 raise ValidationError(ValidationErrorText.VALIDATION_ERROR_AGE_NOT_ACCEPTED)
         return users_birth_date
+
+
+@deconstructible
+class UserDomainEmailValidation:
```

Comment:
If we have now here validation, do we really still need validation in `CustomUserManager` in `_create_user` method?

Reply:
good question. Do we even need `CustomUserManager`?

Reply:
I was wondering whether to delete it, but we have several tests that use this method, so I have to check what it is for and if we need it.

Reply:
I would keep it, it contains two methods which simplifies creation of user and superuser. I would just remove extra validation from there, and add `user.full_check` before `user.save()`.

Reply:
Done

Reply:
What about other validations in `create_user` method? There are two more, the one for `email` could be Django default raised by `full_clean()`, but for password can't because this field is not defined in our `CustomUser` model directly. Maybe we should treat this as special case and allow it either here or in `CustomUser.clean()`? What do you guys think @dybi @kbeker ?

Reply:
But I didn't add to email validators django `EmailValidator` so we are not checking if email is None and then test is going to fail without this validation, I could add this validator, too but then, I my opininon, we would have few same validations twice, like `is email contains "@"`.

Reply:
> What about other validations in `create_user` method? There are two more, the one for `email` could be Django default raised by `full_clean()`, but for password can't because this field is not defined in our `CustomUser` model directly. Maybe we should treat this as special case and allow it either here or in `CustomUser.clean()`? What do you guys think @dybi @kbeker ?

@rwrzesien 
I think that this comming up under special case which we were talking about and should be done in `clean` method in `CustomUser` model :)

Reply:
@Szymiks Why does the test fail exactly? Maybe it need rewriting.

It should fail on `.full_clean()` because the CustomUser model requires `email` to be not-null (`null=True` is the default). So you don't need to set anything extra for this. So in fact what is here https://github.com/Code-Poets/sheetstorm/pull/342/files#diff-d06152d304f72502a60bd8fabb2f792eR47 looks duplicated (unless we need this custom error message, but personally I think that default Django error message for missing field should be enough).

Reply:
Yes It was because of message, django return `This field cannot be blank`